library/Booster/LLVM.hs

```diff
@@ -29,7 +29,7 @@ simplifyTerm api def trm sort = unsafePerformIO $ Internal.runLLVM api $ do
     trmPtr <- Trace.timeIO "LLVM.simplifyTerm.marshallTerm" $ Internal.marshallTerm trm
     sortPtr <- Trace.timeIO "LLVM.simplifyTerm.marshallSort" $ Internal.marshallSort sort
     binary <- liftIO $ kore.simplify trmPtr sortPtr
-    liftIO $ kore.collect
+    liftIO kore.collect
```

Comment:
(off-topic) Interesting that this got accepted by `hlint`  - probably because the `kore.collect` desugars to a class function call (from `HasField`) before `hlint` gets to see it?

Reply:
yeah i was surprised too

Reply:
For the record, `hlint` 3.6 flags it. Perhaps we should consider upgrading from 3.4?